RELEASE_NOTES.md

```diff
@@ -1,6 +1,16 @@
 Release Notes
 =============
 
+v1.0.0-beta.1.x
+---------------
+
+### Breaking changes
```

Comment:
[question] Is this more an improvement as it would segfault before, so not sure if its really "breaking", if you're relying on the segfault then..........

Reply:
As suggested in https://github.com/OpenAssetIO/OpenAssetIO/pull/1193#discussion_r1389264049, changed to "Bug fixes" in 7789fe6b